### PR TITLE
docs(readme.md)[gh-61]: Document supported data types

### PR DESCRIPTION
**Pull request by nbenjamin**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -121,6 +121,111 @@ connection
...
+| [`macaddr8`][psql-macaddr8-ref]                | Not yet supported.|
+| [`money`][psql-money-ref]                      | Not yet supported.|
+| [`numeric`][psql-bignumeric-ref]               | [`BigDecimal`][java-bigdecimal-ref]|
+| [`path`][psql-path-ref]                        | Not yet supported.|
+| [`pg_lsn`][psql-pg_lsn-ref]                    | Not yet supported.|
+| [`point`][psql-point-ref]                      | Not yet supported.|
+| [`polygon`][psql-polygon-ref]                  | Not yet supported.|
+| [`real`][psql-real-ref]                        | [**`Float`**][java-float-ref], [`Double`][java-double-ref]|
+| [`smallint`][psql-smallint-ref]                | [**`Short`**][java-short-ref], [`Boolean`][java-boolean-ref], [`Byte`][java-byte-ref], [`Integer`][java-integer-ref], [`Long`][java-long-ref]|
+| [`smallserial`][psql-smallserial-ref]          | Not yet supported.|
```
Comment (mp911de) [inline README.md]: You might want to recheck on `smallserial` and `serial` (see [`PostgresTestSupport`](https://github.com/spring-projects/spring-data-r2dbc/blob/29f1edce0e42dfe669bf1f71ce5aa06a5cdf4a06/src/test/java/org/springframework/data/r2dbc/testing/PostgresTestSupport.java#L33))

File: README.md [hunk 2]
```diff
@@ -121,6 +121,111 @@ connection
...
+| PostgreSQL Type                                | Supported Data Type                                                                                                                           | 
+|:-----------------------------------------------|:----------------------------------------------------------------------------------------------------------------------------------------------|
+| [`bigint`][psql-bigint-ref]                    | [**`Long`**][java-long-ref], [`Boolean`][java-boolean-ref], [`Byte`][java-byte-ref], [`Short`][java-short-ref], [`Integer`][java-integer-ref] |
+| [`bit`][psql-bit-ref]                          | Not yet supported.|
+| [`bit varying`][psql-bit-ref]                  | Not yet supported.|
+| [`boolean or bool`][psql-boolean-ref]          | [`Boolean`][java-boolean-ref]|
+| [`box`][psql-box-ref]                          | Not yet supported.|
+| [`bytea`][psql-bytea-ref]                      | Not yet supported.|
+| [`character`][psql-character-ref]              | [`String`][java-string-ref]|
+| [`character varying`][psql-character-ref]      | [`String[]`][java-string-ref]|
```
Comment (mp911de) [inline README.md]: `VARCHAR` maps to `String`

File: README.md [hunk 3]
```diff
@@ -121,6 +121,111 @@ connection
...
+| [`time [without time zone]`][psql-time-ref]    | [`LocalTime`][java-lt-ref]|
+| [`time [with time zone]`][psql-time-ref]       | [`LocalTime`][java-lt-ref]|
+| [`timestamp [without time zone]`][psql-time-ref]|[`LocalDateTime`][java-ldt-ref]|
+| [`timestamp [with time zone]`][psql-time-ref]  |[`ZonedDateTime`][java-zdt-ref]|
+| [`tsquery`][psql-tsquery-ref]                  | Not yet supported.|
+| [`tsvector`][psql-tsvector-ref]                | Not yet supported.|
+| [`txid_snapshot`][psql-txid_snapshot-ref]      | Not yet supported.|
+| [`uuid`][psql-uuid-ref]                        | [**`UUID`**][java-uuid-ref], [`String`][java-string-ref]||
+| [`xml`][psql-xml-ref]                          | Not yet supported. |
+
```
Comment (mp911de) [inline README.md]: We should also mention somewhere the array support as array-typed columns are supported for reads and writes.

File: README.md (hunk 1 above)
Comment (nbenjamin) [reply README.md]: I missed those, will update with following 
| [`smallserial`][psql-smallserial-ref]          | [**`Integer`**][java-integer-ref], [`Boolean`][java-boolean-ref], [`Byte`][java-byte-ref], [`Short`][java-short-ref], [`Long`][java-long-ref]|
| [`serial`][psql-serial-ref]                    | [**`Long`**][java-long-ref], [`Boolean`][java-boolean-ref], [`Byte`][java-byte-ref], [`Short`][java-short-ref], [`Integer`][java-integer-ref]|
|

File: README.md (hunk 2 above)
Comment (nbenjamin) [reply README.md]: Changed to `String` instead of `String[]`

File: README.md (hunk 3 above)
Comment (nbenjamin) [reply README.md]: should I add separate section something like this or do you have any suggestion

```
Currently support the following single-dimensional arrays for both read and write

| PostgreSQL Type                                | Supported Data Type                  |
|:-----------------------------------------------|:-------------------------------------|
|[`tex[]`][psql-text-ref]                        |[`String[]`][java-string-ref]         |  
|[`integer[] or int[]`][psql-integer-ref]        |[`Integer[]`][java-integer-ref], [`Long[]`][java-long-ref], [`Short[]`][java-short-ref]|
```

Comment (nbenjamin) [reply README.md]: please let me know if i missed anything in array section

**Comment by nbenjamin** [comment]
@mp911de  please tell me if I missed anything or waiting for anything from me.

File: README.md (hunk 3 above)
Comment (nbenjamin) [reply README.md]: @mp911de  can please let me know if anything missing here or are we good?

**Comment by mp911de** [comment]
Thanks a lot. There are some minor formatting issues that we can handle during the merge.

**Comment by mp911de** [comment]
That's merged and polished now.

**Pull request merged**